test/validation/api/crypto/odp_crypto_test_inp.c

```diff
@@ -407,6 +407,40 @@ static int alg_packet_op_enq(odp_packet_t pkt,
 	return 0;
 }
 
+static int crypto_op(odp_packet_t pkt,
+		     odp_bool_t *ok,
+		     odp_crypto_session_t session,
+		     uint8_t *cipher_iv,
+		     uint8_t *auth_iv,
+		     odp_packet_data_range_t *cipher_range,
+		     odp_packet_data_range_t *auth_range,
+		     uint8_t *aad,
+		     unsigned int hash_result_offset)
+{
+	int rc;
+
+	if (!suite_context.packet)
```

Comment:
The data that is used to choose which operation to call seems to be known already beforehand. Perhaps `alg_test_param_t` could include a function pointer to the relevant operation.

Reply:
The benefit would be that test cases would use the operation function directly instead of a wrapper that fans out to other functions that are chosen based on information that is set elsewhere.